builder/dockerfile/internals.go

```diff
@@ -309,10 +307,31 @@ func (b *Builder) calcCopyInfo(cmdName, origPath string, allowLocalDecompression
 	}
 	origPath = strings.TrimPrefix(origPath, "."+string(os.PathSeparator))
 
+	// validate windows paths from other images
+	if contextID != nil && runtime.GOOS == "windows" {
+		p := strings.ToLower(origPath)
+		if origPath == "/" || strings.HasPrefix(p, "/windows") || strings.HasPrefix(p, "c:/windows") {
```

Comment:
Considering `filepath.FromSlash` couple of lines earlier this is probably wrong.